File: TypeScript/Linked-Lists/IlinkedList.ts
/**
 * An interface for Linked lists, which shares the common methods.
 */
export interface ILinkedList<T> {
  isEmpty(): boolean;
  getIndex(index: number) : T| null| undefined;
  push(data: T): void;
  pop(): T| undefined;
  append(data: T): void;
  removeTail(): T| undefined;
  insertAt(index: number, data: T): void;
  removeAt(index: number): T| undefined;
  clear(): void;
  getLength(): number;
  toArray(): (T| undefined)[];
}

File: Java/LinkedLists/ILinkedList.java
package Java.LinkedLists;

import java.util.List;


public interface ILinkedList<T> {
    public boolean isEmpty();

    public int size();

    public T getIndex(int index);
    
    public void push(T data);

    public T pop();

    public void append(T data);

    public T removeTail();

    public void insertAt(int index, T data);

    public T removeAt(int index);

    public void clear();

    public List<T> toArray();

    public IListNode<T> reverse();
}
